### DISCUSSION TITLE
Interaction tests vs end to end tests with Playwright

### DISCUSSION THREAD

**Discussion by JuAn-Kang** [storybookjs/storybook#29363] [General]
What's the main difference that the two will have? For complex components/pages that are to be made into Stories, it seems like they both offer very similar options, as the interactions addon would allow simulating a series of user events - and also run assertions on it. Are there some distinct pros/cons to either approach given that they are meant to be run on local instances at build time?

**Comment by shilman** [comment]
Play function tests are co-located with your stories and also run in the browser, which means you can easily browse/debug them in Storybooks that are built in CI. The main disadvantage is that testing library uses simulated DOM events which can be less realistic than playwright E2E tests. We will work to address this in the future. This is our recommended workflow.

There's also nothing wrong with writing E2E tests against your Storybook. However, then you have two pieces of code to keep in sync and you lose the convenience of having those tests bundled into your Storybook. But plenty of trans choose this option.
